### PR TITLE
makefile: reduce surprises, NUM_CORES can now be specified on command line

### PR DESCRIPTION
**Pull request by oharboe**
`NUM_CORES=1234  make print-NUM_CORES` now prints 1234

### CODE REVIEW

**Comment by maliberty** [comment]
You could already specify NPROC so now there are two different ways to do the same thing.

**Comment by oharboe** [comment]
> You could already specify NPROC so now there are two different ways to do the same thing.

Yes and the second didn't behave like the first, now it does.

Should one be retired?

**Comment by oharboe** [comment]
Having thought about it: NPROC is the number of threads/CPUs in the system, NUM_CORES is the number of cores ORFS should use.

These are distinct concerns, so I think both variables should be kept.

**Comment by maliberty** [comment]
The only purpose of nproc is to set num cores so it isn't really useful by itself.

**Comment by maliberty** [comment]
I think these should just be collapsed into one var

**Comment by oharboe** [comment]
> I think these should just be collapsed into one var

If I want to use the number of available cores in some reporting or policies, both could be useful, would be my counterargument.

**Comment by oharboe** [comment]
I think the question of whether to collapse these variables is out of scope for this PR.

**Comment by maliberty** [comment]
> I think the question of whether to collapse these variables is out of scope for this PR.

I think have duplicate settable variables is confusing so I think the question is relevant.

> If I want to use the number of available cores in some reporting or policies, both could be useful, would be my counterargument.

We don't do that.  If we did, having NUM_CORES be the number of running threads wouldn't match your idea.  At a minimum things would need renaming and only one should be user settable.

**Comment by oharboe** [comment]
I am happy to eliminate NPROC, will do in this PR 
these things are easily changed when needed.

**Comment by oharboe** [comment]
Seems to work:

```
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ make print-NUM_CORES
NUM_CORES = 20
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ make NUM_CORES=10 print-NUM_CORES
NUM_CORES = 10
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ NUM_CORES=10 make print-NUM_CORES
NUM_CORES = 10
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ echo >>settings.mk export NUM_CORES=10
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ make print-NUM_CORES
NUM_CORES = 10
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ rm settings.mk 
oyvind@small-cigar:~/OpenROAD-flow-scripts/flow$ make print-NUM_CORES
NUM_CORES = 20
```

**Review by maliberty — APPROVED** [review]

File: flow/Makefile [hunk 1]
```diff
@@ -241,26 +241,24 @@ export RTLMP_RPT_FILE ?= partition.txt
 export RTLMP_BLOCKAGE_FILE ?= $(OBJECTS_DIR)/rtlmp/partition.txt.blockage
 
 #-------------------------------------------------------------------------------
-ifeq (, $(strip $(NPROC)))
+ifeq (,$(strip $(NUM_CORES)))
   # Linux (utility program)
-  NPROC := $(shell nproc 2>/dev/null)
+  NUM_CORES := $(shell NUM_CORES 2>/dev/null)
```
Comment (rovinski) [inline flow/Makefile]: `nproc` is a bin util. It can't be replaced here.

Comment (oharboe) [reply flow/Makefile]: Oops. Well spotted. Fixed.

**Comment by oharboe** [comment]
@maliberty Should be good to go now, fixed gaffe spotted by @gadfort

**Comment by maliberty** [comment]
I think you mean spotted by @rovinski

**Pull request merged**